R/ppc-distributions.R

```diff
@@ -528,6 +545,150 @@ ppc_violin_grouped <- function(y, yrep, group, ..., probs = c(0.1, 0.5, 0.9),
     bayesplot_theme_get()
 }
 
+#' @export
+#' @param pit An optional vector of probability integral transformed values for
+#' which the ECDF is to be drawn. If NULL, PIT values are computed to ´y´ with
+#' respect to the corresponding values in ´yrep´.
+#' @rdname PPC-distributions
+#'
+ppc_pit_ecdf <- function(y,
+                         yrep,
+                         ...,
+                         pit = NULL,
+                         K = NULL,
+                         prob = .99,
+                         plot_diff = TRUE,
+                         interpolate_adj = TRUE) {
+  check_ignored_arguments(...,
+      ok_args = c("K", "pit", "prob", "plot_diff", "interpolate_adj"))
+
+  if (is.null(pit)) {
+    pit <- ppc_data(y, yrep) %>%
+      group_by(y_id) %>%
+      dplyr::group_map(~ mean(.x$value[.x$is_y] >= .x$value[!.x$is_y])) %>%
+      unlist()
+    if (is.null(K)) {
+      K <- min(length(unique(ppc_data(y, yrep)$rep_id)) + 1, length(pit))
+    }
+  } else {
+    inform("'pit' specified so ignoring 'y', and 'yrep' if specified.")
+    pit <- validate_pit(pit)
+    if (is.null(K)) {
+      K <- length(pit)
+    }
+  }
+  N <- length(pit)
+  gamma <- adjust_gamma(N,
+                        K = K,
+                        prob = prob,
+                        interpolate_adj = interpolate_adj)
+  lims <- ecdf_intervals(N, K = K, gamma = gamma)
+  ggplot() +
+    aes_(
+      x = 1:K / K,
+      y = ecdf(pit)(seq(0, 1, length.out = K)) - (plot_diff == TRUE) * 1:K / K,
+      color = "y"
+    ) +
+    geom_step(show.legend = FALSE) +
+    geom_step(aes(
+      y = lims$upper[-1] / N - (plot_diff == TRUE) * 1:K / K,
+      color = "yrep"
+    ), show.legend = FALSE) +
+    geom_step(aes(
+      y = lims$lower[-1] / N - (plot_diff == TRUE) * 1:K / K,
+      color = "yrep"
+    ), show.legend = FALSE) +
+    yaxis_title(FALSE) +
+    xaxis_title(FALSE) +
+    yaxis_ticks(FALSE) +
+    scale_color_ppc_dist() +
+    bayesplot_theme_get()
+}
+
+#' @export
+#' @rdname PPC-distributions
+#'
+ppc_pit_ecdf_grouped <-
+  function(y,
+           yrep,
+           group,
+           ...,
+           K = NULL,
+           pit = NULL,
+           prob = .99,
+           plot_diff = TRUE,
+           interpolate_adj = TRUE) {
+    check_ignored_arguments(...,
+      ok_args = c("K", "pit", "prob", "plot_diff", "interpolate_adj"))
+
+    if (is.null(pit)) {
+      pit <- ppc_data(y, yrep, group) %>%
+        group_by(y_id) %>%
+        dplyr::group_map(~ mean(.x$value[.x$is_y] >= .x$value[!.x$is_y])) %>%
+        unlist()
+      if (is.null(K)) {
+        K <- length(unique(ppc_data(y, yrep)$rep_id)) + 1
+      }
+    } else {
+      inform("'pit' specified so ignoring 'y', and 'yrep' if specified.")
```

Comment:
I don't understand the reason for the "," behind 'y'

Reply:
No reason, just a typo. It has been removed.